piksi_multi_rtk_gps/src/geodetic_survey.py

```diff
@@ -66,25 +69,35 @@ def navsatfix_spp_callback(self, msg):
 
     def set_base_station_position(self, lat0, lon0, alt0):
         everything_ok = []
-        rospy.loginfo("Setting Piksi Multi surveyed position to: %.10f, %.10f, %.1f" % (lat0, lon0, alt0))
+        # Use only 2 digits for altitude as it is more than enough.
+        alt0 = float("%.2f" % alt0)
+        rospy.loginfo("Setting Piksi Multi surveyed position to: %.10f, %.10f, %.1f\n" % (lat0, lon0, alt0))
 
         # Write settings.
         write_lat0_ok = self.write_settings_to_piksi("surveyed_position", "surveyed_lat", "%.10f" % lat0)
         write_lon0_ok = self.write_settings_to_piksi("surveyed_position", "surveyed_lon", "%.10f" % lon0)
-        write_alt0_ok = self.write_settings_to_piksi("surveyed_position", "surveyed_alt", "%.10f" % alt0)
+        write_alt0_ok = self.write_settings_to_piksi("surveyed_position", "surveyed_alt", "%.2f" % alt0)
 
         if write_lat0_ok and write_lon0_ok and write_alt0_ok:
             # Save and check what was actually written to flash
             settings_saved = self.save_settings_to_piksi()
             if settings_saved:
-                read_lat0 = self.read_settings_from_piksi("surveyed_position", "surveyed_lat")
-                read_lon0 = self.read_settings_from_piksi("surveyed_position", "surveyed_lon")
-                read_alt0 = self.read_settings_from_piksi("surveyed_position", "surveyed_alt")
+                read_lat0, read_lat0_value = self.read_settings_from_piksi("surveyed_position", "surveyed_lat")
+                read_lon0, read_lon0_value = self.read_settings_from_piksi("surveyed_position", "surveyed_lon")
+                read_alt0, read_alt0_value = self.read_settings_from_piksi("surveyed_position", "surveyed_alt")
 
                 if read_lat0 and read_lon0 and read_alt0:
-                    everything_ok = True
-                    self.log_surveyed_position(lat0, lon0, alt0)
+                    # Check read values == computed values
+                    if self.is_close(float(read_lat0_value), lat0) and self.is_close(float(read_lon0_value),
+                                                                                     lon0) and self.is_close(
+                            float(read_alt0_value), alt0):
+                        everything_ok = True
+                        self.log_surveyed_position(lat0, lon0, alt0)
+                    else:
+                        rospy.logwarn("Read values do NOT correspond with written ones. Please use piksi conole.")
```

Comment:
typo: "[...] correspond **to** written ones. Please use piksi con**s**ole
Maybe also add in the warning what to use console for: to us it's clear, what we should do with the console, to a new user not necessarily.